kratos/tests/cpp_tests/input_output/test_stl_io.cpp

```diff
@@ -90,4 +92,47 @@ endsolid 3 triangles
     KRATOS_CHECK_EQUAL(r_model_part.GetSubModelPart("3 triangles").NumberOfElements(), 3);
 }
 
+KRATOS_TEST_CASE_IN_SUITE(WriteTriangleToSTL, KratosCoreFastSuite)
+{
+    Model current_model;
+    ModelPart& r_model_part = current_model.CreateModelPart("Main");
+
+    // create a few nodes
+    r_model_part.CreateNewNode(1, 0.1, -2.56114e-08, 0.1);
+    r_model_part.CreateNewNode(2, 0.1, -0.499156, -0.0352136);
+    r_model_part.CreateNewNode(3, 0.1, -0.473406, -0.0446259);
+
+    // create a triangle element
+    Properties::Pointer p_properties(new Properties(0));
+    r_model_part.CreateNewElement("Element3D3N", 101, {1, 2, 3}, p_properties);
+
+    // write stl
+    std::string filename = "test_stl_write.stl";
+    StlIO * stl_write = new StlIO(filename, IO::WRITE);
```

Comment:
Instead of the manual memory allocation you should out those lines into a scope, which will have the same effect

Reply:
thanks for the suggestion!